runbotics/runbotics-scheduler/src/websocket/bot/bot-lifecycle.service.ts

```diff
@@ -1,62 +1,110 @@
-                            newProcessInstance.error
-                        }"`
-                    );
-                })
-                .catch((err) => {
-                    this.logger.log(`<= Error: process-instance (${newProcessInstance.id}) update after bot disconnection failed | Error: ${err}`);
-                });
-
-            this.handleInterruptedProcessInstanceEvent(bot, processInstance.id);
+            await this.updateInterruptedProcessInstance(newProcessInstance, processInstance.id);            
+
+            await this.handleProcessInstanceEventInterruption(processInstance, bot);
         });
     }
 
-    private async handleInterruptedProcessInstanceEvent (bot: IBot, processInstanceId: string): Promise<void> {
-        const activeProcessInstanceEvents = this.processInstanceEventService.findActiveByProcessInstanceId(processInstanceId);
-        const newProcessInstanceEvent = {
-            ...activeProcessInstanceEvents,
-            status: ProcessInstanceEventStatus.ERRORED,
-            error: 'Bot has been shut down',
-        };
+    private async handleProcessInstanceEventInterruption (processInstance: IProcessInstance, bot: IBot): Promise<void> {
+        const activeEvents: IProcessInstanceEvent[] = await this.processInstanceEventService.findActiveByProcessInstanceId(processInstance.id);
+        const processId: number = await this.processInstanceService.findById(processInstance.id).then(processInstance => processInstance.process.id);
+        
+        if(!activeEvents) return;
+        
+        activeEvents.forEach(async event => {
+            const newProcessInstanceEvent: IProcessInstanceEvent = {
+                ...event, 
+                status: ProcessInstanceEventStatus.ERRORED, 
+                error: 'Bot has been shut down',
+                processInstance: {
+                    id: processInstance.id,
+                    process: {
+                        id: processId
+                    }
+                }
+            };
+            
+            await this.updateInterruptedProcessInstanceEvent(newProcessInstanceEvent, bot);
+        });
+    }
+
+    private async updateInterruptedProcessInstance(processInstance: IProcessInstance, installationId: string) {
+        this.logger.log(
+            `<= Success process-instance (${
+                processInstance.id
+            }) has been updated after bot disconnection | status: ${
+                processInstance.status
+            }, error message: "${
+                processInstance.error
+            }"`
+        );
+
+        this.botProcessService.updateProcessInstance(installationId, processInstance);
+
+        this.logger.log(
+            `=> Updating process-instance (${
+                processInstance.id
+            }) updated after bot disconnection | status: ${
+                processInstance.status
+            }, error message: "${
+                processInstance.error
+            }"`
+        );
```

Comment:
there is a misconception. This kind of logs with arrows are meant to mean the incoming and outgoing info but here there are other operations

Reply:
changed, take a look